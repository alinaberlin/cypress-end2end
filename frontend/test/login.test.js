import { mount, createLocalVue } from '@vue/test-utils'
import Login from '@/pages/login.vue'
import Vuex from 'vuex'
const localVue = createLocalVue()

localVue.use(Vuex)

describe('Login', () => {
  let actions
  let store

  beforeEach(() => {
    actions = {
      login: jest.fn(),
    }

    store = new Vuex.Store({
      actions,
    })
  })

  test('Login page', () => {
    const wrapper = mount(Login, {
      stubs: ['router-link'],
    })
    expect(wrapper).toMatchSnapshot()
  })

  test('Login user', async () => {
    const routerMock = { push: jest.fn() }

    const wrapper = mount(Login, {
      stubs: ['router-link'],
      mocks: { $router: routerMock },
      store,
      localVue,
    })

    const email = wrapper.find('#email')
    email.element.value = 'armagan@coyotiv.com'
    email.trigger('input')

    const password = wrapper.find('#password')
    password.element.value = 'test'
    password.trigger('input')

    await wrapper.find('form').trigger('submit')

    await expect(actions.login).toHaveBeenCalledWith(expect.anything(), {
      email: 'armagan@coyotiv.com',
      password: 'test',
    })

    expect(routerMock.push).toHaveBeenCalledWith('/profile')
  })
})
