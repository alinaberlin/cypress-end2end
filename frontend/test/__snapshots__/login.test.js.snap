// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Login page 1`] = `
<div class="login">
  <form>
    <h1>Log in to your account</h1><label for="email">Email:&nbsp;<input id="email" type="email" placeholder="Your email" required="required"></label><label for="password">Password:&nbsp;<input id="password" type="password" placeholder="Your password" required="required"></label><input type="submit" value="Log in">
  </form>
  <!---->
  <div>Don't have an account yet? <router-link-stub to="/register">Register</router-link-stub>
  </div>
</div>
`;
